packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -867,8 +867,10 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
   ui.FrameInfo? _nextFrame;
   // When the current was first shown.
   late Duration _shownTimestamp;
-  // The requested duration for the current frame;
+  // The requested duration for the current frame.
   Duration? _frameDuration;
+  // The default duration for the frame that _frameDuration is 0.
+  static const Duration _DefaultDuration = Duration(milliseconds: 67);
```

Comment:
Where did this magic number 67 come from? Please provide in a comment how you chose this number. (I assume it was meant as a closest integer to 1000 ms / 15 FPS.)

Reply:
I found in this Android gif plugin that the default delay time is 100ms：

https://github.com/koral--/android-gif-drawable/blob/c61ede92d53b096f4fe8c6b00c98e725102c0f3b/android-gif-drawable/src/main/c/gif.h#L42

https://github.com/koral--/android-gif-drawable/blob/b23ff5b456256e56df7ceb088da341e530f6942c/android-gif-drawable/src/main/c/decoding.c#L157

But I found that the playback speed is slower than my mac preview, I estimated a delay time similar to the mac preview is 67ms.

Reply:
Number 67 seems reasonable, but please leave the comment in code (e.g. "eyeballed to match GIF preview speed on macOS {os version}") so that when another developer reads the code after you knows where this number came from. Also, it might be useful to add that this delay produces cca. 15 FPS.

Reply:
We should find a standard that defines this (probably one of the WHATWG standards), so that we can match other implementations, rather than guessing or eyeballing or just copying other code bases.

Reply:
@Hixie 
I haven't found any standard that supports this method, but I have seen similar methods in many gif parsing codes;

Glide：
https://github.com/bumptech/glide/blob/d2bb3e8a6a2296b20ecaeb7759c16ef15aacbb28/third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/GifHeaderParser.java#L117

YLGIFImage:
https://github.com/liyong03/YLGIFImage/blob/4b953716ca600c50a95445ef1f0bb5bdc0eb878a/YLGIFImage/YLGIFImage.m#L41

SDWebImage:
https://github.com/SDWebImage/SDWebImage/blob/15e16e9e7f2511112a4c0b71f3d59988adbdc3ad/SDWebImage/Core/SDImageIOAnimatedCoder.m#L177

Their comments have links to explanations, but some of them are invalid; If you know a similar standard, please let me know, thank you very much.

Reply:
I agree with Hixie. I assumed there is a some standard as well, but could not find it after a quick search. It is possible that this case is similar to the case of frame rate in movies, some use 30, 60, 29.97, or 59.94.

Reply:
@PollexLee you've collected quite a lot of information here about how this is done by lots of implementors. It might be a good idea to file a but on the WHATWG folks to document this, since there's so much evidence. (And the WebP thing in particular should probably be done in the WebP spec.)